tests/staking/inflation.test.ts

```diff
@@ -46,6 +46,14 @@ given('astar')('Reward payouts based on inflation (decayed)', async ({ networks:
     }
 
     const targetBlockHeight = currentBlock + remainingBlocks
+
+    // Set the current block number in storage to be exactly 1 less than target
+    // This satisfies the runtime's assertion: current_block + 1 == new_block
+    await astar.dev.setStorage({
+      system: {
+        number: targetBlockHeight - 1,
+      },
+    })
     await astar.dev.newBlock({ count: 1, unsafeBlockHeight: targetBlockHeight })
```

Comment:
I'm surprised that `.newBlock` doesn't handle this update since it will fail otherwise (except if advancing to the next consecutive block).

Reply:
Me too, even when using the latest chopstick version. The new assert check in the frame system code is new (1month old) and the issue is happening when using `unsafeBlockHeight` so maybe it has not been patched yet? I will consider open an issue later to flag it.

Reply:
newBlock doesn't handle it because there's no assertion, or at least there was not